crates/l2/sdk/src/sdk.rs

```diff
@@ -382,19 +367,55 @@ pub enum DeployError {
     ProxyBytecodeNotFound,
 }
 
-pub async fn deploy_contract(
+pub async fn create_deploy(
+    client: &EthClient,
+    deployer: &Signer,
+    init_code: Bytes,
+    overrides: Overrides,
+) -> Result<(H256, Address), EthClientError> {
+    let mut deploy_overrides = overrides;
+    deploy_overrides.to = Some(TxKind::Create);
+
+    let deploy_tx = build_generic_tx(
+        client,
+        TxType::EIP1559,
+        Address::zero(),
+        deployer.address(),
+        init_code,
+        deploy_overrides,
+    )
+    .await?;
+    let deploy_tx_hash = send_generic_transaction(client, deploy_tx, deployer).await?;
+
+    let nonce = client
+        .get_nonce(deployer.address(), BlockIdentifier::Tag(BlockTag::Latest))
+        .await?;
+    let mut encode = vec![];
+    (deployer.address(), nonce).encode(&mut encode);
+
+    //Taking the last 20bytes so it matches an H160 == Address length
+    let deployed_address = Address::from_slice(keccak(encode).as_fixed_bytes().get(12..).ok_or(
+        EthClientError::Custom("Failed to get deployed_address".to_owned()),
+    )?);
+
+    wait_for_transaction_receipt(deploy_tx_hash, client, 1000).await?;
+
+    Ok((deploy_tx_hash, deployed_address))
+}
+
+pub async fn create2_deploy_from_path(
     constructor_args: &[u8],
     contract_path: &Path,
     deployer: &Signer,
     salt: &[u8],
     eth_client: &EthClient,
 ) -> Result<(H256, Address), DeployError> {
     let bytecode = hex::decode(read_to_string(contract_path)?)?;
-    deploy_contract_from_bytecode(constructor_args, &bytecode, deployer, salt, eth_client).await
+    create2_deploy_from_bytecode(constructor_args, &bytecode, deployer, salt, eth_client).await
 }
 
 /// Same as `deploy_contract`, but takes the bytecode directly instead of a path.
```

Comment:
I think the comment got outdated

```suggestion
/// Same as `create2_deploy_from_path `, but takes the bytecode directly instead of a path.
```

Or we can remove it since the funtion name already says what it is in the comment